src/LanguageServer/Impl/Sources/DefinitionSource.cs

```diff
@@ -206,7 +206,7 @@ internal sealed class DefinitionSource {
                 case IPythonClassType cls:
                     // Data members may be instances that are not tracking locations.
                     // In this case we'll try look up the respective variable instead.
-                    using (eval.OpenScope(analysis.Document, cls.ClassDefinition)) {
+                    using (eval.OpenScope(cls.DeclaringModule, cls.ClassDefinition)) {
```

Comment:
There's one other place which doesn't look like this when opening a scope for a class definition, in `AssignmentHandler.TryHandleClassVariable`:

```
Eval.OpenScope(Eval.Module, cls.ClassDefinition, out _)
```

Is that one also wrong?

```
$ rg 'OpenScope.*Class'
LanguageServer\Impl\Sources\DefinitionSource.cs
209:                    using (eval.OpenScope(analysis.Document, cls.ClassDefinition)) {

Analysis\Ast\Impl\Extensions\PythonClassExtensions.cs
31:                using (eval.OpenScope(cls.DeclaringModule, cls.ClassDefinition)) {

Analysis\Ast\Impl\Analyzer\Evaluation\ExpressionEval.Callables.cs
93:                using (OpenScope(cls.DeclaringModule, cls.ClassDefinition, out _)) {

Analysis\Ast\Impl\Analyzer\Handlers\AssignmentHandler.cs
99:                    using (Eval.OpenScope(Eval.Module, cls.ClassDefinition, out _)) {
```

Reply:
No, that one is OK since it actually handles `self` which is the module the eval is going through. However, I can make an extension method to reduce confusion which would accept type and always use proper module from the type.

Reply:
Whatever works, I just did a quick grep because I was curious if there were any more left.